File: leetcode/Easy/0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.cpp
/*
 Problem: 28. Find the Index of the First Occurrence in a String
 Language: cpp
 Runtime: 0 ms (100.00%)
 Memory: 8.5 MB (99.97%)
 Tags: Two Pointers, String, String Matching
*/
class Solution {
public:
    int strStr(string haystack, string needle) {
        int n = haystack.size();
        int m = needle.size();

        if (m == 0)
            return 0;

        for (int i = 0; i <= n - m; ++i) {
            bool match = true;
            for (int j = 0; j < m; ++j) {
                if (haystack[i + j] != needle[j]) {
                    match = false;
                    break;
                }
            }
            if (match)
                return i;
        }

        return -1;
    }
};

File: leetcode/Easy/0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.py
"""
 Problem: 28. Find the Index of the First Occurrence in a String
 Language: python3
 Runtime: 36 ms (5.07%)
 Memory: 16.6 MB (99.99%)
 Tags: Two Pointers, String, String Matching
"""
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        return haystack.find(needle)